src/dune_rules/cm_files.mli

```diff
@@ -15,9 +15,9 @@ val make :
   -> unit
   -> t
 
-val unsorted_objects_and_cms : t -> mode:Mode.t -> Path.t list
+val unsorted_objects_and_cms : t -> mode:Lib_mode.t -> Path.t list
 
-val top_sorted_cms : t -> mode:Mode.t -> Path.t list Action_builder.t
+val top_sorted_cms : t -> mode:Lib_mode.t -> Path.t list Action_builder.t
 
 val top_sorted_objects_and_cms :
-  t -> mode:Mode.t -> Path.t list Action_builder.t
+  t -> mode:Lib_mode.t -> Path.t list Action_builder.t
```

Comment:
From what I recall, these functions are only needed for linking executables and building archives. Neither of those tasks can be done with melange, so I believe the old types should remain here.